### ISSUE TITLE
Add option for overriding scheduler step

### ISSUE THREAD

**Issue by benihime91** [PyTorchLightning/pytorch-lightning#5059]
### Motivation
Currently there is no way to overide the `scheduler.step()`.  `scheduler.step()` calls for all the schedulers are called sequentially i.e,
```python
for scheduler in schedulers:
    scheduler.step()
```
There should be something a hook that enables to override the `scheduler.step()` call something similar to the existing hook for optimizers `LightningModule.optimizer_step()`

### Pitch
something similar to :
```python
def scheduler_step(self, epoch, batch_idx, scheduler, scheduler_idx, *args, **kwargs)
    if scheduler_idx == 0:
       ....   # do something
       scheduler.step()
    
    if scheduler_idx == 1:
       ....   # do something here
       scheduler.step()
 ....
```

**Comment by benihime91** [comment]
@tchaton can you have a look as discussed in slack ?

**Comment by rohitgr7** [comment]
just thinking, should it be given in the arguments itself within `optimizer_step`??

**Comment by benihime91** [comment]
Something like this probably wherein `scheduler` and `scheduler_idx` are passed as arguments to `optimizer_step`

```python
def optimizer_step(current_epoch, batch_nb, optimizer, optimizer_idx, scheduler, scheduler_idx, *args, **kwargs):
      pass
```

**Comment by rohitgr7** [comment]
will scheduler_idx be useful here? wouldn't it always be equal to `optimizer_idx`??

**Comment by benihime91** [comment]
Now that you mention it , that seems to the case so only `optmizer_idx` should suffice.
